torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -292,6 +296,23 @@ at::Tensor inferAndAlloc(
         ") for the buffer ",
         tv->toString());
     inferred_sizes.push_back(inferred_val.value());
+    if (inferred_val.value() == 1 &&
+        expanded_map.count(expanded_sizes.size())) {
+      auto expanded_size = expanded_map.at(expanded_sizes.size());
+      const auto inferred_expanded_size = expr_eval.evaluate(expanded_size);
+      TORCH_INTERNAL_ASSERT(
+          inferred_expanded_size.has_value(),
+          "Could not launch kernel as program could not infer the expanded extent ",
+          expanded_size->toString(),
+          "(",
+          expanded_size->name(),
+          ") for the buffer ",
+          tv->toString());
+      expanded_sizes.push_back(inferred_expanded_size.value());
+      expanded_dim = true;
+    } else {
+      expanded_sizes.push_back(inferred_val.value());
```

Comment:
Is it an error if `expanded_map` has an entry for this size if its inferred size is not one?

Reply:
They should strictly be the same which would be fine, but I made `expanded_extent` meaningless if it's not on a broadcast dimension.

Reply:
Added a constraint to make sure if this is the case `inferred_size` and `expanded_extent` are evaluated to the same value.